torch/nested/_internal/ops.py

```diff
@@ -929,20 +930,49 @@ def unbind_int(func, *args, **kwargs):
     lengths = inp.lengths()
     ragged_idx = inp._ragged_idx
 
+    def _torch_check(_lengths: List[int], _offsets: Optional[List[int]] = None):
```

Comment:
cc @jbschlosser @soulitzer for the NJT changes

Reply:
I'm good with these. It's also super exciting that we can use `unbind()` now within compile without graph breaks, as long as `torch._dynamo.config.capture_scalar_outputs = True` is set.

one minor nit would be to add some small amount of documentation describing the reason for this logic. (i.e. to provide torch.compile() with the information it's unable to obtain itself regarding the relationship among unbacked SymInts that result from the data-dependency on offsets)

Reply:
Yes, _torch_check() info was needed to bypass symbolic shapes during split/unbind. Will add a comment.